### ISSUE TITLE
bug(python): ColbertReranker uses wrong architecture & other inconsistencies

### ISSUE THREAD

**Issue by AyushExel** [lancedb/lancedb#1546]
### LanceDB version

v5.0

### What happened?

- ColBERT predates the `trust_remote_code` option on HF, so the model architecture is actually wrong 
- MaxSim works by summing rather than averaging -- the torch.mean() at the end will mess things up (you should just sum up the score of all query tokens)
- The query length is super important, and needs a specific kind of padding with <MASK> token to take query augmentation into account, but not too many <MASK>s, as they're just augmenters.
-
